src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -282,7 +286,8 @@ SHAMapStoreImp::run()
     while (1)
     {
         healthy_ = true;
-        validatedLedger_.reset();
+        validatedLedger.reset();
```

Comment:
What do you think of ea3e8c4? It's pretty similar to what you wrote, but makes use of a `std::pair` in the return result instead of `wasStopped`.